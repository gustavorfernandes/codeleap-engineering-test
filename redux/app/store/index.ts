import { configureStore } from "@reduxjs/toolkit"
import usernameReducer from "../slices/usernameSlice"

export const store = configureStore({
  reducer: {
    username: usernameReducer
  },
});

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = ReturnType<typeof store.dispatch>
