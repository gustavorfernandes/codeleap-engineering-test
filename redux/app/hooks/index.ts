import { Dispatch } from "@reduxjs/toolkit"
import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux"
import { RootState } from "../store"

export const useAppDispatch = () => useDispatch<Dispatch>()
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector