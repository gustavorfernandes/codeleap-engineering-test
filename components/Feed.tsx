/* eslint-disable react-hooks/exhaustive-deps */
import PostCard from "./PostCard"
import { IPost } from "../types/IPost"
import { useAppSelector, useAppDispatch } from "../redux/app/hooks"
import { orderByDate, setPostList } from "../redux/app/slices/postListSlice"
import EditModal from "./EditModal"
import DeleteAlert from "./DeleteAlert"
import getPosts from "../actions/getPosts"
import { useEffect } from "react"

function Feed() {
  const dispatch = useAppDispatch()

  useEffect(() => {
    getPosts().then((res) => {
      const data = res.results
      dispatch(setPostList(data))
    })
    dispatch(orderByDate())    
  })

  const postList = useAppSelector((state) => state.postList.value)

  return (
    <>
      <div className="w-screen max-w-[800px] bg-white flex flex-col items-center">
        {postList &&
          postList.map((post: IPost) => (
            <>
              <PostCard key={post.id} id={post.id} title={post.title} created_datetime={post.created_datetime} content={post.content} username={post.username} />

              <DeleteAlert />

              <EditModal />
            </>
          ))}
      </div>
    </>
  )
}

export default Feed
