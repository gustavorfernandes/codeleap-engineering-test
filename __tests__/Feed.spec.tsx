import React from "react"
import "@testing-library/jest-dom"
import { fireEvent, render, screen } from "@testing-library/react"
import { Provider } from "react-redux"
import { store } from "../redux/app/store"
import Feed from "../components/Feed"

describe("Feed component", () => {
  it("When the component starts, it should list the posts", () => {
    render(
      <Provider store={store} >
        <Feed />
      </Provider>
    )
    //teste
  })

  it("The component must render correctly", () => {
    const { container } = render(
      <Provider store={store} >
        <Feed />
      </Provider>
    )
    expect(container).toMatchSnapshot()
  })
})
