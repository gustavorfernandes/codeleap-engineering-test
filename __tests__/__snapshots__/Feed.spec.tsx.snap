// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed component The component must render correctly 1`] = `
<div>
  <div
    class="w-screen max-w-[800px] bg-white flex flex-col items-center"
  />
</div>
`;
